docs/TeslaSleepConfiguration.md

```diff
@@ -0,0 +1,118 @@
+
+Using Bluetooth hints, a tool like tasker on the phone can then detect the Bluetooth connection between the app and the vehicle, and send a hint to TeslaMate to anticipate that the vehicle will wake up, and to resume high-frequency polling. The following sections of the document deal with different phone platforms and how to set up bluetooth hints for those platforms.
+
+#### Notes
+
+  * If you are specifying the local (LAN) IP address/port of your TeslaMate instance when performing the wake up, this may fail for a number of reasons:
+     * It will not provide hints to TeslaMate when you are outside of your home network (ie if you have been driving).
+     * It will not provide hints to TeslaMate if the mobile phone becomes disconnected from the wifi network.
+  * For these reasons, it is highly recommended that the hints be provided over a public interface to TeslaMate, which may require that interface to 
+
+#### Android
+
+There are a number of applicatons that provide this functionality on Android:
+
+| Application | Price   | Guide(s) |
+| ----------- |---------|----------|
+| [Automagic](https://play.google.com/store/apps/details?id=ch.gridvision.ppam.androidautomagic&hl=en_AU)   | US$3.50 | N/A |
+| MacroDroid  | Free (Limited)   | Testing |
+| Tasker      | US$3.00 | [TaskerSetup](TaskerSetup.md) |
+
+#### iPhone
+
+On the iOS platform, Apple provides the Shortcuts workflow tool:
+
+| Application | Price   | Guide(s) |
+| ----------- |---------|----------|
+| [Shortcuts](https://apps.apple.com/us/app/shortcuts/id915249334)   | Free    | Coming Soon |
+
+### Event-Based Hints
+
+In addition to Bluetooth hints, which would address most use cases, it is also possible to institute event-based hints. An example of an event-based hint is to trigger a callback to TeslaMate when my Garage Door is opened via HomeAssistant. The relevant configuration for this is:
+
+  * automations.yaml
+```
+- alias: Wake Teslamate on Garage Open
+  initial_state: on
+  trigger:
+    - platform: state
+      entity_id: switch.garage_door_switch
+      to: 'off'
+  action:
+    - service: script.turn_on
+      entity_id: script.wake_teslamate
+```
+
+  * script.yaml
+```
+wake_teslamate:
+  sequence:
+    - service: shell_command.wake_teslamate
+
+sleep_teslamate:
+  sequence:
+    - service: shell_command.sleep_teslamate
+```
+
+  * shell_command.yaml
+
+_Note: Substitute your IP address for the commands_
+```
+  sleep_teslamate: curl -X PUT http://192.168.1.1:4000/api/car/1/logging/suspend
+  wake_teslamate: curl -X PUT http://192.168.1.1:4000/api/car/1/logging/resume
+```
+
+### Time of Day Hints
+
+_Are these available? I think this would be a good addition to TeslaMate, if not_
```

Comment:
Could you elaborate on this?

Reply:
Just a thought I'd had while writing the doc - a feature request that I was considering would be the ability to specify times (for example, 7am - 8am daily) where TeslaMate would resume polling in anticipation of the vehicle waking up. This could help for daily commuters who aren't able to get a stable setup using Tasker, etc. What are your thoughts?